unit-tests-java8/src/test/java/org/eclipse/collections/test/stack/StackIterableTestCase.java

```diff
@@ -99,36 +100,36 @@ default void RichIterable_getFirst_empty_null()
     @Test
     default void RichIterable_getLast_empty_null()
     {
-        assertThrows(UnsupportedOperationException.class, this.newWith()::getLast);
+        assertThrows(EmptyStackException.class, () -> this.newWith().getLast());
```

Comment:
This is fine for now. @donraab what do you think of changing the behavior to match other collections? pop() would still throw, but getFirst() and getLast() would return null.

Reply:
I'm kind of conflicted with having `getLast()` for Stack. Stack is both LIFO and FILO. So what should getFirst and getLast mean in the context of Stack? I see that the implementation calls delegate.getFirst() for ArrayStack. I think getFirst and getLast should probably behave the same as peek in terms of throwing or not throwing.

Reply:
I think overrides in tests are a hint that the interface hierarchy has a problem. We have getFirst and getLast on RichIterable and they return null on empty RichIterables unless:

- if you call getFirst() on an empty stack, you get an exception indicating it's an empty stack
- if you call getLast() on an empty stack, either the same thing or an UnsupportedOperationException because you're only supposed to look at the front of a stack

Even on other collection types, when getFirst() returns null, we don't know from the returned value alone whether the collection is empty or has a null at the front. I tend to think we should deprecate these methods across the board. They're already deprecated on RichIterable, but overridden to be not deprecated on OrderedIterable. We're talking here about a case where they behave inconsistently on an OrderedIterable.